src/main/scala/com/adobe/api/platform/runtime/mesos/MesosClient.scala

```diff
@@ -90,7 +90,8 @@ case class TaskDef(taskId: String,
                    network: Network = Bridge,
                    dockerRunParameters: Map[String, Set[String]] = Map.empty,
                    commandDef: Option[CommandDef] = None,
-                   constraints: Set[Constraint] = Set.empty)
+                   constraints: Set[Constraint] = Set.empty,
+                   healthCheckParams: Map[String, Int] = Map.empty)
```

Comment:
Instead of a `Map` here, please create a case class to encapsulate the config, as well as allow disabling of health checks. I suggest something like:
```
case class HealthCheckConfig(delay, interval, timeout, gracePeriod, maxConsecutiveFailures)
```
and the TaskDef sig should be 
```
healthCheckConfig: Option[HealthCheckConfig]
``` 
then the config will be well encapsulated, as well as optional

Reply:
Thx @tysonnorris . I will put in the case class. When the healthcheckportindex is not specified, we are not enabling the health check. Does that suffice for "allowing disabling of health check" ? That way, we can put in the default health check params if, only healthcheckportindex is specified and not health check params. What do you think ?

Reply:
In `MesosClient`, I think `healthCheckPortIndex: Option[Int] = None` should be replaced by `healthCheckConfig: Option[HealthCheckConfig] = None`, so the whole healthcheckconfig becomes optional.